lib/chef/provider/google_key_pair.rb

```diff
@@ -162,16 +163,10 @@ def action_handler
     @action_handler ||= Chef::Provisioning::ChefProviderActionHandler.new(self)
   end
 
-  # TODO de-dup with driver or abstract to base class
-  # TODO update to take the full operation body
-  def wait_for_operation(operation_id)
-    Retryable.retryable(:tries => 30, :sleep => 5, :matching => /Not done/) do |retries, exception|
-      action_handler.report_progress("  waited #{retries*5}/#{30*5}s for operation #{operation_id} to complete")
-      # TODO it is really awesome that there are 3 types of operations, and the only way of telling
-      # which is which is to parse the full `:selfLink` from the response body
-      operation = driver.operations_client.global_get(operation_id)
-      raise "Not done" unless operation[:body][:status] == "DONE"
-      operation
+  def reupload_metadata(metadata)
+    converge_by 'reuploading changed metadata' do
+      operation = driver.projects_client.set_common_instance_metadata(metadata)
```

Comment:
Fixed. I don't know what exactly I did, I thought I ran it and it worked, but apparently I ran something else... But now it works definitely.